### PR TITLE
nwfilter: Remove dependency on autotest

### PR DESCRIPTION
**Pull request by waynesun09**
Remove dependency on autotest and use avocado.

Signed-off-by: Wayne Sun <gsun@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 1]
```diff
@@ -83,16 +83,17 @@ def run(test, params, env):
...
+            process.run(cmd, shell=True)
 
         if ipset_command:
             try:
-                os_dep.command("ipset")
-            except ValueError:
-                ret = utils.run("yum install ipset -y")
+                path.find_command("ipset")
+            except path.CmdNotFoundError:
+                ret = process.run("yum install ipset -y", shell=True)
```
Comment (nasastry) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: For `RPM` based machines the above `yum install ipset -y` will work with out any issues. How about machines which has package manager like `deb`?

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 2]
```diff
@@ -83,16 +83,17 @@ def run(test, params, env):
...
                 if ret.exit_status:
-                    raise error.TestNAError("Can't install ipset on host")
-            utils.run(ipset_command)
+                    raise exceptions.TestSkipError("Can't install ipset on
```
Comment (nasastry) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: You might see an error like following
`exceptions.TestSkipError("Can't install ipset on`
`SyntaxError: EOL while scanning string literal`

**Comment by waynesun09** [comment]
This depends on:
https://github.com/avocado-framework/avocado-vt/pull/942

File: libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_define.py [hunk 3]
```diff
@@ -96,42 +95,25 @@ def run(test, params, env):
...
-            if status:
-                err_msg = "Run failed with right command."
-                if bug_url:
-                    err_msg += " Check more info in %s" % bug_url
-                raise error.TestFail(err_msg)
+        utlv.check_exit_status(cmd_result, status_error)
+        if not status_error:
+            # Check result
+            chk_result = check_list(filter_uuid, filter_name)
+            xml_path = "%s/%s.xml" % (NWFILTER_ETC_DIR, filter_name)
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_define.py]: `os.path.join()` can be used.

File: libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_list.py [hunk 4]
```diff
@@ -43,25 +44,19 @@ def run(test, params, env):
...
         # Retrieve filter name from output and check the cfg file
         output_list = output.split('\n')
-        for i in range(2, len(output_list)):
-            filter_name.append(output_list[i].split()[1])
-        for i in range(len(filter_name)):
-            xml_path = "%s/%s.xml" % (NWFILTER_ETC_DIR, filter_name[i])
+        for line in output_list[1:]:
+            filter_name.append(line.split()[1])
+        for name in filter_name:
+            xml_path = "%s/%s.xml" % (NWFILTER_ETC_DIR, name)
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_list.py]: `os.path.join()` can be used here too

File: libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_list.py [hunk 5]
```diff
@@ -32,8 +33,8 @@ def run(test, params, env):
 
     if not libvirt_version.version_compare(1, 1, 1):
         if params.get('setup_libvirt_polkit') == 'yes':
-            raise error.TestNAError("API acl test not supported in current"
-                                    " libvirt version.")
+            raise exceptions.TestSetupSkip("API acl test not supported in "
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_list.py]: in other files, exceptions.`TestSkipError()` is used, here we use `TestSetupSkip()`.

File: libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py [hunk 6]
```diff
@@ -68,22 +68,23 @@ def run(test, params, env):
         if "DEVNAME" in check_cmd:
             check_cmd = check_cmd.replace("DEVNAME", iface_target)
```
Comment (balamuruhans) [inline libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py]: should `L58` to be handled,
`iface_target = iface_xml.target['dev']`
```
virttest.libvirt_xml.xcepts.LibvirtXMLNotFoundError: Error in Getter for property "target", element tag "target" not found on parent at xpath "/" in XML
<?xml version='1.0' encoding='UTF-8'?>
<interface type="bridge">
      <mac address="52:54:00:3e:3f:40" />
      <source bridge="virbr0" />
      <model type="virtio" />
      <filterref filter="allow-arp" />
      <address bus="0x00" domain="0x0000" function="0x0" slot="0x01" type="pci" />
    </interface>

```

Comment (waynesun09) [inline libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py]: @balamuruhans 
https://libvirt.org/formatdomain.html#elementsNICSTargetOverride

If no target is specified, certain hypervisors will automatically generate a name for the created tun device. This name can be manually specified, however the name should not start with either 'vnet' or 'vif', which are prefixes reserved by libvirt and certain hypervisors. Manually specified targets using these prefixes may be ignored.

so by default without specifying target, after start the vm, libvirt will create with certain name target tun device.

We did not get this error, and in the case vm is started first before to get the target name.

File: libvirt/tests/src/nwfilter/nwfilter_edit_uuid.py [hunk 7]
```diff
@@ -53,15 +52,17 @@ def run(test, params, env):
...
+        except (aexpect.ShellError, aexpect.ExpectError,
+                remote.LoginTimeoutError), details:
             log = session.get_output()
             session.close()
             if "Try again? [y,n,f,?]:" in log and status_error:
                 logging.debug("edit uuid failed as expected.")
             else:
-                raise error.TestFail("Failed to do nwfilter-edit: %s\n%s"
-                                     % (details, log))
+                raise exceptions.TestFail("Failed to do nwfilter-edit: "
```
Comment (balamuruhans) [inline libvirt/tests/src/nwfilter/nwfilter_edit_uuid.py]: let us use this opportunity for start using test.fail() and respective changes elsewhere

File: libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py (hunk 6 above)
Comment (balamuruhans) [inline libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py]: @waynesun09 thank you for clarifying 👍

File: libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py [hunk 8]
```diff
@@ -68,22 +67,23 @@ def run(test, params, env):
...
         ret = utils_misc.wait_for(lambda: not
-                                  utils.system(check_cmd, ignore_status=True),
+                                  process.system(check_cmd,
+                                                 ignore_status=True),
                                   timeout=30)
         if not ret:
-            raise error.TestFail("Rum command '%s' failed" % check_cmd)
-        out = utils.system_output(check_cmd, ignore_status=False)
+            test.fail("Rum command '%s' failed" % check_cmd)
```
Comment (balamuruhans) [inline libvirt/tests/src/nwfilter/nwfilter_update_vm_running.py]: `%s/Rum/Run`

File: libvirt/tests/src/nwfilter/nwfilter_vm_attach.py [hunk 9]
```diff
@@ -68,23 +67,22 @@ def run(test, params, env):
...
             ret = utils_misc.wait_for(lambda: not
-                                      utils.system(check_cmd,
-                                                   ignore_status=True),
+                                      process.system(check_cmd,
+                                                     ignore_status=True),
                                       timeout=30)
             if not ret:
-                raise error.TestFail("Rum command '%s' failed" % check_cmd)
-            out = utils.system_output(check_cmd, ignore_status=False)
+                test.fail("Rum command '%s' failed" % check_cmd)
```
Comment (balamuruhans) [inline libvirt/tests/src/nwfilter/nwfilter_vm_attach.py]: same here `%s/Rum/Run`

File: libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_list.py [hunk 10]
```diff
@@ -42,26 +40,18 @@ def run(test, params, env):
...
         # Retrieve filter name from output and check the cfg file
         output_list = output.split('\n')
-        for i in range(2, len(output_list)):
-            filter_name.append(output_list[i].split()[1])
-        for i in range(len(filter_name)):
-            xml_path = "%s/%s.xml" % (NWFILTER_ETC_DIR, filter_name[i])
+        for line in output_list[2:]:
+            filter_name.append(line.split()[1])
+        for name in filter_name:
+            xml_path = "%s/%s.xml" % (NWFILTER_ETC_DIR, name)
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/filter/virsh_nwfilter_list.py]: `os.path.join()` can be used to concat path

**Pull request closed without merging**